test/controllers/resource_controller_test.rb

```diff
@@ -60,4 +60,55 @@ class ResourceControllerTest < ActionDispatch::IntegrationTest
       end
     end
   end
+
+  test 'should allow #create when recaptcha is enabled and recaptcha is verified' do
+    @api_namespace.api_form.update(show_recaptcha: true)
+    payload = {
+      data: {
+          properties: {
+            first_name: 'Don',
+            last_name: 'Restarone'
+          }
+      }
+    }
+    assert_difference "@api_namespace.api_resources.count", +1 do
+      post api_namespace_resource_index_url(api_namespace_id: @api_namespace.id), params: payload
```

Comment:
how do we stub the passing recapcha? It looks like the recapcha validation is only enforced on the client-side

Reply:
I see the reason now, recapcha passes by default in test env: https://stackoverflow.com/questions/24011793/recaptcha-in-rails-verify-recaptcha-returning-always-true-in-test?rq=1

Reply:
Recaptcha is valid in test env by default. To test the invalid condition I had to add this:
```
Recaptcha.configuration.skip_verify_env.delete("test")
```

Reply:
@Pralish yup! Makes sense now, thanks for the clarification